build-system/tasks/check-build-system.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+const {execOrThrow} = require('../common/exec');
+
+/**
+ * Performs type checking on the /build-system directory using TypeScript.
+ * Configuration is defined in /build-system/tsconfig.json.
+ *
+ */
+function checkBuildSystem() {
+  execOrThrow(
+    'npx -p typescript tsc --project ./build-system/tsconfig.json',
+    'TypeScript build failed'
+  );
+}
+
+checkBuildSystem.description = 'Check the code in /build-system for errors.';
```

Comment:
nit: For clarity, let's follow the convention in `check-types.js`:

https://github.com/ampproject/amphtml/blob/fb7f095545bb43cab153b7629114ced95f7ed715/build-system/tasks/check-types.js#L128

```suggestion
checkBuildSystem.description = 'Check source code in build-system/ for JS type errors.';
```